pkg/controller/pravega/pravega_segmentstore.go

```diff
@@ -329,18 +333,31 @@ func MakeSegmentStoreHeadlessService(pravegaCluster *api.PravegaCluster) *corev1
 
 func MakeSegmentStoreExternalServices(pravegaCluster *api.PravegaCluster) []*corev1.Service {
 	var service *corev1.Service
+	var ssPodName string
+	var ssFQDN string
+
+	domainName := strings.TrimSpace(pravegaCluster.Spec.ExternalAccess.DomainName)
 	services := make([]*corev1.Service, pravegaCluster.Spec.Pravega.SegmentStoreReplicas)
 
 	for i := int32(0); i < pravegaCluster.Spec.Pravega.SegmentStoreReplicas; i++ {
+		ssPodName = util.ServiceNameForSegmentStore(pravegaCluster.Name, i)
+		if strings.HasSuffix(domainName, dot) {
+			ssFQDN = ssPodName + dot + domainName
+		} else {
+			ssFQDN = ssPodName + dot + pravegaCluster.Spec.ExternalAccess.DomainName + dot
+		}
 		service = &corev1.Service{
 			TypeMeta: metav1.TypeMeta{
 				Kind:       "Service",
 				APIVersion: "v1",
 			},
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      util.ServiceNameForSegmentStore(pravegaCluster.Name, i),
+				Name:      ssPodName,
 				Namespace: pravegaCluster.Namespace,
 				Labels:    util.LabelsForSegmentStore(pravegaCluster),
+				Annotations: map[string]string{
+					externalDNSAnnotationKey: ssFQDN,
```

Comment:
To eliminate the default suffix of `pravega.io`, compute an `annotations` map above, which may or may not contain an external DNS annotation.   Then, on this line, you can simply use the map.

Reply:
I think we would need to have a default 'domainname' value to handle the case where external access is enabled but domainname is not set by the user in the manifest, especially since this value is also going to be used in "init_kubernetes.sh" for advertising SS service. The other option is to throw an error when that happens.

Reply:
Ok.